packages/engine-core/src/binary/BinaryEngine.ts

```diff
@@ -944,18 +949,24 @@ You very likely have the wrong "binaryTarget" defined in the schema.prisma file.
 
   async requestBatch<T>(
     queries: EngineQuery[],
-    { headers = {}, transaction, numTry = 1, containsWrite }: RequestBatchOptions,
+    { traceparent, transaction, numTry = 1, containsWrite }: RequestBatchOptions<undefined>,
   ): Promise<BatchQueryEngineResult<T>[]> {
     await this.start()
 
-    const request: QueryEngineBatchRequest = {
-      batch: queries.map(({ query }) => ({ query, variables: {} })),
-      transaction: Boolean(transaction),
-      isolationLevel: transaction?.isolationLevel,
+    const headers: Record<string, string> = {}
+    if (traceparent) {
+      headers.traceparent = traceparent
     }
 
+    const itxId = getInteractiveTransactionId(transaction)
```

Comment:
```suggestion
    const itx = transaction?.kind === 'itx' ? transaction.options : undefined
```
(Optional, for consistency with dataproxy)